crates/sqlexec/src/context/local.rs

```diff
@@ -320,6 +324,28 @@ impl LocalSessionContext {
         self.portals.remove(name);
     }
 
+    pub(crate) fn push_notice(&mut self, notice: Notice) {
+        self.notices.push(notice)
+    }
+
+    /// Drain all notices from the session.
+    ///
+    /// This will take into account the 'client_min_messages' session var to
+    /// filter out notices that shouldn't be sent to the client.
+    pub(crate) fn drain_notices(&mut self) -> Vec<Notice> {
+        // Behind an if since this is called every query, and session vars is behind  a lock :)
+        // Let's get rid of the lock.
```

Comment:
does this mean that we're constructing an empty vector for every iteration?

Reply:
Yes, but empty vecs don't allocate.

Reply:
wat. ok.